plugin/core/registry.py

```diff
@@ -122,6 +122,19 @@ def session(self) -> Optional[Session]:
         else:
             return None
 
+    def sessions(self) -> List[Session]:
```

Comment:
I wonder whether this should better be a generator, like the method with the same name in LspTextCommand? Just to make it consistent with the other class.

Reply:
It was using generators in this PR before I've changed.

The reason I've changed it is that I don't believe this is a proper use case for generators since iterating over the sessions and returning a small list is very cheap. I haven't measured the performance of generators but it must be a lot more expensive since essentially those would end up with a function call to get each Session.

Generators should IMO be used when iteration is expensive and when there is benefit to allowing the caller to interrupt the iteration at any point. Shouldn't just blindly be used like they are the cool or something. :)

If we want to be consistent then I'd rather change the other place to also return a list.

Reply:
Well, the one in LspTextCommand is actually used from one place in a way that takes advantage of generators: https://github.com/sublimelsp/LSP/blob/07ef2a05e55034a222e6f2f4da13b463a51ac417/plugin/formatting.py#L50-L53 so I wouldn't want to change that.

Reply:
for  consistency reasons I gave up and changed back to generator

Reply:
Haha, okay I didn't look at the separate commits but only the file diff. I fully agree to your reasoning. But I also favor the consistency more in this case.